### PR TITLE
Bump dependencies to supported libraries only

### PR DESCRIPTION
**Pull request by tacman**
| Q | A
| --- | ---
| Branch? | 3.0

I took a stab at upgrading the dependencies.

amazonwebservices/aws-sdk-for-php is abandoned, and is the only reason we have to support psr/log ^2.  I'm pretty sure we can remove it in this version, as mentioned elsewhere it's only for V1 of the cache.  WDYT?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,37 +22,36 @@
...
-        "symfony/console": "^5.3|^6.0|^7.0",
-        "symfony/dependency-injection": "^5.3|^6.0|^7.0",
-        "symfony/form": "^5.3|^6.0|^7.0",
-        "symfony/messenger": "^5.3|^6.0|^7.0",
-        "symfony/phpunit-bridge": "^5.3|^6.0|^7.0",
-        "symfony/templating": "^5.3|^6.0|^7.0",
-        "symfony/validator": "^5.3|^6.0|^7.0",
-        "symfony/yaml": "^5.3|^6.0|^7.0"
+        "psr/cache": "^3.0",
+        "psr/log": "^1.0|^2.0|^3,0",
```
Comment (dbu) [inline composer.json]:
```suggestion
        "psr/log": "^1.0|^2.0|^3.0",
```

File: README.md [hunk 2]
```diff
@@ -226,37 +218,18 @@ web_profiler :
 
 ### Runtime Options
 
-Sometime, you may have a filter defined that fulfills 99% of your usage
-scenarios. Instead of defining a new filter for the erroneous 1% of cases,
-you may instead choose to alter the behavior of a filter at runtime by
+Sometime, you may may need to modify your filter at runtime.  You can do so by
```
Comment (dbu) [inline README.md]: yeah, agree this is less verbose :+1: 

```suggestion
Sometime, you may may need to modify your filter at runtime. You can do so by 
```

File: README.md [hunk 3]
```diff
@@ -323,15 +289,12 @@ lifting for you.
...
-class MyController extends Controller
+class MyController extends AbstractController
 {
-    public function indexAction()
-    {
-        /** @var FilterService */
-        $imagine = $this
-            ->container
-            ->get('liip_imagine.service.filter');
+use Liip\ImagineBundle\Service\FilterService;
```
Comment (dbu) [inline README.md:291]: this should go above the class declaration

**Review by dbu — APPROVED** [review]
thanks a lot!

yeah i agree, lets remove support for the abandoned aws sdk version.

File: README.md (hunk 3 above)
Comment (tacman) [reply README.md:291]: yeah, that's what I get for writing PHP directly in the README!

File: .github/workflows/phpunit.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        php: ['8.0', '8.1', '8.2']
+        php: ['8.1', '8.2', '8.3']
```
Comment (dbu) [inline .github/workflows/phpunit.yml:18]: there seems to be one php 8.0 build left according to the CI

File: composer.json [hunk 5]
```diff
@@ -22,37 +22,36 @@
...
-        "symfony/cache": "^5.3|^6.0|^7.0",
-        "symfony/console": "^5.3|^6.0|^7.0",
-        "symfony/dependency-injection": "^5.3|^6.0|^7.0",
-        "symfony/form": "^5.3|^6.0|^7.0",
-        "symfony/messenger": "^5.3|^6.0|^7.0",
-        "symfony/phpunit-bridge": "^5.3|^6.0|^7.0",
-        "symfony/templating": "^5.3|^6.0|^7.0",
-        "symfony/validator": "^5.3|^6.0|^7.0",
-        "symfony/yaml": "^5.3|^6.0|^7.0"
+        "psr/cache": "^3.0",
```
Comment (dbu) [inline composer.json:43]: this breaks the symfony 5.4 install. but as this is a new version, i think we should just go with symfony 6.4|7.0 only. the 2.x branch is fine and did not see tons of changes, so people stuck on symfony 5 can keep using that.

**Comment by dbu** [comment]
great, this is shaping up well.

can you please add a changelog entry explaining what got removed and what to use instead?
please remove the build for symfony 5.4 and the one with php 8.0

i fixed the lowest build with php 8.2 in the 3.x branch, rebasing on 3.x should fix it here too.

**Comment by tacman** [comment]
I took a stab at fixing the phpunit workflow, but not 100% sure I got right.

> i fixed the lowest build with php 8.2 in the 3.x branch, rebasing on 3.x should fix it here too.

I didn't see that, and I'll admit that I'm not sure how to rebase.

**Comment by dbu** [comment]
git rebase means that you tell git to update your branch with the 3.x branch and apply your commits at the end of that branch. if you have checked out your fork as `origin`, you could do `git remote add liip git@github.com:liip/LiipImagineBundle.git` and `git remote update` to have upstream available, then while you are on your branch, you do `git rebase liip/3.x` and then `git push --force-with-lease`. alternatively you can use git merge to merge the 3.x branch into your branch. that will create a merge commit, but i can squash the commits while merging so that is fine too.

**Comment by tacman** [comment]
OK, I'll try that.  What I've been doing is going to the github page and
syncing when it says my branch is behind, then pulling to pull local
machine.

But I should learn rebase.

**Comment by dbu** [comment]
constant learning :+1: 

but i don't want to block your progress, if its too complicated i can do the rebase to keep things moving.

**Comment by tacman** [comment]
Yes, please.

Also, can you provide a working version of the flysytem bundle config and
liip_imagine config?  I can provide a simple repo if that helps.

I have a messier repo, too, that used to work with Oneup but now won't
install, it's a 2-year-old project and I'm fighting with dependencies.  Ugh.

**Comment by dbu** [comment]
merged in #1557

i don't have a testing repository around with flysystem

**Pull request closed without merging**